cmds/chmod/chmod.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2016 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// sync command in Go.
+package main
+
+import (
+	"flag"
+	"log"
+	"strconv"
+	"syscall"
+)
+
+func main() {
+
+	flag.Parse()
+	if len(flag.Args()) < 2 {
+		flag.PrintDefaults()
+		log.Fatalf("usage: chmod mode filepath")
+	}
+
+	var mode uint32
+
+	if octval, err := strconv.ParseUint(flag.Args()[0], 8, 32); err != nil {
+		log.Fatalf("Unable to decode mode. Please use a hex value. arg was %s, err was %v", flag.Args()[0], err)
+	} else if octval > 0777 {
+		log.Fatalf("Invalid octal value. Value larger than 777, was %o", octval)
+	} else {
+		mode = uint32(octval)
+	}
+
+	for i := 1; i < len(flag.Args()); i++ {
+		if err := syscall.Chmod(flag.Args()[i], mode); err != nil {
+			log.Fatalf("Unable to chmod, filename was %s, err was %v", flag.Args()[i], err)
```

Comment:
suggestion here. It's useful to see all the errors, not just the first one. Maybe you could do this: 
var errors string
before the for
then in here you do a ok += fmt.Sprintf("error ...")
then at the end, check if errrors is not ""  and if so log.Fatalf(errors)
and otherwise just fall through to the end.